src/routes/users/postUsers.ts

```diff
@@ -0,0 +1,54 @@
+import express, { Request, Response, NextFunction } from 'express';
+import db from '../../knex';
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+import { validate, setLocales, en } from 'robust-validator';
+
+const router = express.Router();
+const JWT_SECRET = process.env.JWT_SECRET || 'key';
+const saltRounds = 10;
+
+// Validation setup
+setLocales(en);
+
+const createUserValidation = {
+  email: 'required|email',
+  password: 'required|min:6|max:50',
+  full_name: 'required|min:3|max:100', 
+};
+
+router.post('/', async (req: Request, res: Response, next: NextFunction) => {
+  const { email, password, full_name } = req.body; 
+
+  // Validation
+  const result = await validate({ email, password, full_name }, createUserValidation);
+
+  if (Object.keys(result.errors).length > 0) {
+    return res.status(400).json({ message: result.errors });
+  }
+
+  try {
+    const existingUser = await db('users').where({ email }).first();
+
+    if (existingUser) {
+      return res.status(400).json({ message: 'Email already exists' });
+    }
+
+   
+    const hashedPassword = await bcrypt.hash(password, saltRounds);
```

Comment:
Create a `src/constants.ts` file, and put the `saltRounds` value in there. You will use the same value in the login in future. Variable name should be something like `SALT_ROUNDS`